### PR TITLE
CMake: record git information in a separate configuration file

### PR DESCRIPTION
**Pull request by tamiko**

### CODE REVIEW

File: doc/news/changes/incompatibilities/20230703MatthiasMaier [hunk 1]
```diff
@@ -0,0 +1,12 @@
+Changed: The CMake configuration does no longer export git revision strings
+and dates in `deal.IIConfig.cmake`. This avoids unnecessary
+reconfigurations of client projects (such as the testsuite). Instead the
+git revision strings and dates are now recording in
```
Comment (tjhei) [inline doc/news/changes/incompatibilities/20230703MatthiasMaier:4]:
```suggestion
git revision strings and dates are now recorded in
```

File: doc/news/changes/incompatibilities/20230703MatthiasMaier [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+`deal.IIConfigGit.cmake` whose path is recorded in `${DEAL_II_GIT_CONFIG}`.
+This is similar to how we store the git revision in the header
+`deal.II/base/revision.h` instead of `deal.II/base/config.h` to avoid
+unnecessary recompilations of the library and user programs. User project
+who need the information in cmake must do an
```
Comment (tjhei) [inline doc/news/changes/incompatibilities/20230703MatthiasMaier:9]:
```suggestion
who need the information in CMake must do an
```

**Review by tjhei — APPROVED** [review]

**Comment by tamiko** [comment]
@tjhei Updated.

**Review by masterleinad — APPROVED** [review]

**Pull request merged**